fitlins/cli/run.py

```diff
@@ -60,94 +60,102 @@ def get_parser():
     g_bids.add_argument('--space', action='store',
-                        choices=['MNI152NLin2009cAsym', 'T1w'], default='MNI152NLin2009cAsym',
+                        choices=['MNI152NLin2009cAsym', 'T1w'],
                         help='registered space of input datasets')
 
     g_perfm = parser.add_argument_group('Options to handle performance')
-    g_perfm.add_argument('--debug', action='store_true', default=False,
+    g_perfm.add_argument('--debug', action='store_true',
                          help='run debug version of workflow')
-    g_perfm.add_argument('--nthreads', '--n_cpus', '-n-cpus', action='store', default=0, type=int,
+    g_perfm.add_argument('--nthreads', '--n_cpus', '-n-cpus', action='store', type=int,
                          help='maximum number of threads across all processes')
-    g_perfm.add_argument('--omp-nthreads', action='store', type=int, default=0,
+    g_perfm.add_argument('--omp-nthreads', action='store', type=int,
                          help='maximum number of threads per-process')
-    g_perfm.add_argument('--mem_mb', '--mem-mb', action='store', default=0, type=int,
+    g_perfm.add_argument('--mem_mb', '--mem-mb', action='store', type=int,
                          help='upper bound memory limit for FMRIPREP processes')
-    g_perfm.add_argument('--use-plugin', action='store', default=None,
+    g_perfm.add_argument('--use-plugin', action='store',
                          help='nipype plugin configuration file')
 
     g_other = parser.add_argument_group('Other options')
-    g_other.add_argument('-w', '--work-dir', action='store', default='work',
+    g_other.add_argument('-w', '--work-dir', action='store',
                          help='path where intermediate results should be stored')
     g_other.add_argument(
-        '--reports-only', action='store_true', default=False,
+        '--reports-only', action='store_true',
         help='only generate reports, don\'t run workflows. This will only rerun report '
              'aggregation, not reportlet generation for specific nodes.')
     g_other.add_argument(
-        '--run-uuid', action='store', default=None,
+        '--run-uuid', action='store',
         help='Specify UUID of previous run, to include error logs in report. '
              'No effect without --reports-only.')
-    g_other.add_argument('--write-graph', action='store_true', default=False,
+    g_other.add_argument('--write-graph', action='store_true',
                          help='Write workflow graph.')
 
     return parser
 
 
-def main():
+def main(args=None):
     """Entry point"""
+    if args == None:
+        args=sys.argv[1:]
+
     warnings.showwarning = _warn_redirect
-    opts = get_parser().parse_args()
-    if opts.debug:
-        logger.setLevel(logging.DEBUG)
+    opts = get_parser().parse_args(args)
+    opts = {k:v for k,v in vars(opts).items() if v is not None} # Remove None
 
-    create_workflow(opts)
+    create_workflow(**opts)
```

Comment:
I think I'd rather replace the whole PR with:

```Diff
- def main():
+ def main(args=None):
...
-     opts = get_parser.parse_args()
+     opts = get_parser.parse_args(args)
```

I think this does what you actually want, without doing a lot of refactoring now. (`parse_args` default value for `args` is already `None`, so there's no real need for checking values and branching.)

While there are a couple things I can point to that I think I would probably end up reverting in a refactor, most of my objection boils down to "this is a lot of change in code for a very small change in functionality".